src/lib/falcor/FalcorController.js

```diff
@@ -103,15 +101,65 @@ export default class FalcorController extends BaseComponent {
         });
       }
       else {
-        const err = new Error("FalcorPathSets: " + JSON.stringify(falcorPathSets) + " returned no data.")
-        console.error(err);
+        console.error("FalcorPathSets: " + JSON.stringify(falcorPathSets) + " returned no data.");
         this.safeSetState({
           ready: true,
           fetching: false,
           data: null,
-          error: err,
         });
       }
+      let globalError = getGlobalError();
+      if (globalError && globalError.message === "Response code 0") {
+        resetGlobalError();
+      }
+    }).catch((err) => {
+      try {
+        if (!(err instanceof Array)) {
+          if (!(err instanceof Error)) {
+            throw new Error("Unexpected Falcor Error. Caught error looks like this after JSON.stringify: " + JSON.stringify(err));
+          }
+          else {
+            err.message = "Error comes from falcor fetch: " + err.message;
+            throw err;
+          }
+        }
+        else {
+          let S = new Set();
```

Comment:
In general I think this is handling multiple errors in the wrong place. You should be handling multiple errrors from within the error library / file. And the components causing the errors should not need to manually deal with multiple errors like this. What if two errors came from completely different parts of the program?